### DISCUSSION TITLE
Calling Route Handler on Preview Deployment

### DISCUSSION THREAD

**Discussion by benjibauss** [vercel/community#7020] [Help] [answered]
### Summary

I would like my team to be able to test out an API I created with NextJS route handlers which is on a Vercel Preview deployment. Every time I call a route e.g "GET /api/foo" it returns the HTML for the Vercel login page.

### Steps to Reproduce

Deploy some api route handlers to a Preview deployment and try accessing them via Postman or Insomnia or any HTTP client.

**Comment by QuiiBz** [comment, marked as answer]
You can bypass the deployment protection in several ways, but since you mentioned using an HTTP client, you might want to use the `x-vercel-protection-bypass` header: https://vercel.com/docs/security/deployment-protection/methods-to-bypass-deployment-protection/protection-bypass-automation#protection-bypass-for-automation

**Reply by benjibauss** [reply]
Thank you very much for your quick response! However I just found out that I would need to pay 150$ for that feature.

**Reply by QuiiBz** [reply]
Quick update: we just made an update to allow using `x-vercel-protection-bypass` for free, on all plans: https://vercel.com/changelog/improved-security-with-automation-testing-now-available-on-all-plans
